staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go

```diff
@@ -166,7 +165,7 @@ type Interface interface {
 }
 
 // Manufacture will create a lock of a given type according to the input parameters
-func New(lockType string, ns string, name string, coreClient corev1.CoreV1Interface, coordinationClient coordinationv1.CoordinationV1Interface, rlc ResourceLockConfig) (Interface, error) {
+func New(lockType string, ns string, name string, coordinationClient coordinationv1.CoordinationV1Interface, rlc ResourceLockConfig) (Interface, error) {
```

Comment:
This seems to be a public API. Is it safe to change it this way?
@pohly WDYT?

Reply:
+1

Reply:
The `coreClient` param  is no longer used in this function

Reply:
Merging this PR would break real-world code:
https://grep.app/search?q=resourcelock.New&filter[path][0]=pkg/

We should not change this API unless there is a very strong reason for it. "Cleaning up" is not strong enough.

Reply:
This is a public API that may be used by others downstream. Any modifications could introduce risks. We do not want to introduce risks for users. In my understanding, altering API parameters should be approached with caution. If it's just about cleaning up unused variables, I don't think this change is necessary. Unless we have a specific change in mind for this API, it's best to remove this variable "incidentally" as part of a clearly defined modification.

Reply:
Got it. Thanks